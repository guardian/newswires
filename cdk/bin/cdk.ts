import 'source-map-support/register';
import { RiffRaffYamlFile } from '@guardian/cdk/lib/riff-raff-yaml-file';
import { App } from 'aws-cdk-lib';
import { STACK } from '../../shared/constants';
import type { PollerId } from '../../shared/pollers';
import { pollerIdToLambdaAppName, POLLERS_CONFIG } from '../../shared/pollers';
import { Newswires } from '../lib/newswires';
import { WiresFeeds } from '../lib/wires-feeds';

const app = new App();

const env = {
	region: 'eu-west-1',
};

const codeWiresFeeds = new WiresFeeds(app, 'WiresFeeds-CODE', {
	env,
	stack: STACK,
	stage: 'CODE',
});

const prodWiresFeeds = new WiresFeeds(app, 'WiresFeeds-PROD', {
	env,
	stack: STACK,
	stage: 'PROD',
});

new Newswires(app, 'Newswires-CODE', {
	env,
	stack: STACK,
	stage: 'CODE',
	domainName: 'newswires.code.dev-gutools.co.uk',
	enableMonitoring: false,
	sourceQueue: codeWiresFeeds.sourceQueue,
	fingerpostQueue: codeWiresFeeds.fingerpostQueue,
}).addDependency(codeWiresFeeds);

new Newswires(app, 'Newswires-PROD', {
	env,
	stack: STACK,
	stage: 'PROD',
	domainName: 'newswires.gutools.co.uk',
	enableMonitoring: false,
	sourceQueue: prodWiresFeeds.sourceQueue,
	fingerpostQueue: prodWiresFeeds.fingerpostQueue,
}).addDependency(prodWiresFeeds);

export const riffraff = new RiffRaffYamlFile(app);

const pollerLambdaIds = (Object.keys(POLLERS_CONFIG) as PollerId[]).map(
	pollerIdToLambdaAppName,
);
riffraff.riffRaffYaml.deployments.forEach((deployment) => {
	if (
		deployment.type === 'aws-lambda' &&
		pollerLambdaIds.includes(deployment.app)
	) {
		deployment.contentDirectory = 'poller-lambdas';
	}
});

riffraff.synth();
