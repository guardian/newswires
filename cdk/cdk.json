{
	"app": "ts-node-dev bin/cdk.ts",
	"context": {
		"aws-cdk:enableDiffNoFail": "true",
		"@aws-cdk/core:stackRelativeExports": "true"
	}
}
