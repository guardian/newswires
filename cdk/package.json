{
	"name": "cdk",
	"version": "0.0.0",
	"private": true,
	"scripts": {
		"build": "tsc",
		"test": "jest --detectOpenHandles --config ../jest.config.js --selectProjects cdk",
		"test-update": "jest -u",
		"synth": "cdk synth --path-metadata false --version-reporting false",
		"diff": "cdk diff --path-metadata false --version-reporting false",
		"format": "prettier --write \"{lib,bin}/**/*.ts\"",
		"lint": "eslint lib/** bin/** --ext .ts --no-error-on-unmatched-pattern --fix",
		"lint:ci": "eslint lib/** bin/** --ext .ts --no-error-on-unmatched-pattern"
	},
	"devDependencies": {
		"@guardian/cdk": "59.3.0",
		"@guardian/prettier": "^8.0.1",
		"aws-cdk": "2.153.0",
		"aws-cdk-lib": "2.153.0",
		"constructs": "10.3.0",
		"prettier": "^3.3.3",
		"source-map-support": "0.5.20",
		"ts-node": "^10.9.2"
	},
	"prettier": "@guardian/prettier",
	"jest": {
		"testMatch": [
			"<rootDir>/lib/**/*.test.ts"
		],
		"transform": {
			"^.+\\.tsx?$": "ts-jest"
		},
		"setupFilesAfterEnv": [
			"./jest.setup.js"
		]
	}
}
