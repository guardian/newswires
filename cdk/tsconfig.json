{
	"extends": "../tsconfig.json",
	"include": ["lib/**/*", "bin/**/*"],
	"exclude": ["node_modules", "cdk.out", "lib/**/__snapshots__/**", "dist"],
	"compilerOptions": {
		"module": "ES2020",
		"moduleResolution": "node"
	}
}
