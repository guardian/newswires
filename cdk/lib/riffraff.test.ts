import { riffraff } from '../bin/cdk';

describe('generated riff-raff.yaml', () => {
	it('matches the snapshot', () => {
		// @ts-ignore
		const outdir = riffraff.outdir; // this changes for every test execution and best not to change cdk.ts too much
		const riffRaffYaml = riffraff.toYAML().replaceAll(outdir, 'cdk.out');
		expect(riffRaffYaml).toMatchSnapshot();
	});
});
