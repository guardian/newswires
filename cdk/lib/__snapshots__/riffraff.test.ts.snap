// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`generated riff-raff.yaml matches the snapshot 1`] = `
"allowedStages:
  - CODE
  - PROD
deployments:
  cfn-eu-west-1-editorial-feeds-wires-feeds:
    type: cloud-formation
    regions:
      - eu-west-1
    stacks:
      - editorial-feeds
    app: wires-feeds
    contentDirectory: cdk.out
    parameters:
      templateStagePaths:
        CODE: WiresFeeds-CODE.template.json
        PROD: WiresFeeds-PROD.template.json
  lambda-upload-eu-west-1-editorial-feeds-fingerpost-queueing-lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: fingerpost-queueing-lambda
    contentDirectory: fingerpost-queueing-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: fingerpost-queueing-lambda.zip
    actions:
      - uploadLambda
  lambda-upload-eu-west-1-editorial-feeds-ingestion-lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: ingestion-lambda
    contentDirectory: ingestion-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: ingestion-lambda.zip
    actions:
      - uploadLambda
  lambda-upload-eu-west-1-editorial-feeds-reingestion-initiation-lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: reingestion-initiation-lambda
    contentDirectory: reingestion-initiation-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: reingestion-initiation-lambda.zip
    actions:
      - uploadLambda
  lambda-upload-eu-west-1-editorial-feeds-email-filter-lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: email-filter-lambda
    contentDirectory: email-filter-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: email-filter-lambda.zip
    actions:
      - uploadLambda
  lambda-upload-eu-west-1-editorial-feeds-cleanup-lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: cleanup-lambda
    contentDirectory: cleanup-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: cleanup-lambda.zip
    actions:
      - uploadLambda
  lambda-upload-eu-west-1-editorial-feeds-reuters_poller_lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: reuters_poller_lambda
    contentDirectory: poller-lambdas
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: poller-lambdas.zip
    actions:
      - uploadLambda
  lambda-upload-eu-west-1-editorial-feeds-apPoller_poller_lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: apPoller_poller_lambda
    contentDirectory: poller-lambdas
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: poller-lambdas.zip
    actions:
      - uploadLambda
  asg-upload-eu-west-1-editorial-feeds-newswires:
    type: autoscaling
    actions:
      - uploadArtifacts
    regions:
      - eu-west-1
    stacks:
      - editorial-feeds
    app: newswires
    parameters:
      bucketSsmLookup: true
      prefixApp: true
    contentDirectory: newswires
  cfn-eu-west-1-editorial-feeds-newswires:
    type: cloud-formation
    regions:
      - eu-west-1
    stacks:
      - editorial-feeds
    app: newswires
    contentDirectory: cdk.out
    parameters:
      templateStagePaths:
        CODE: Newswires-CODE.template.json
        PROD: Newswires-PROD.template.json
      amiParametersToTags:
        AMINewswires:
          BuiltBy: amigo
          AmigoStage: PROD
          Recipe: editorial-tools-jammy-java17
          Encrypted: 'true'
    dependencies:
      - lambda-upload-eu-west-1-editorial-feeds-fingerpost-queueing-lambda
      - lambda-upload-eu-west-1-editorial-feeds-ingestion-lambda
      - lambda-upload-eu-west-1-editorial-feeds-reingestion-initiation-lambda
      - lambda-upload-eu-west-1-editorial-feeds-email-filter-lambda
      - lambda-upload-eu-west-1-editorial-feeds-cleanup-lambda
      - lambda-upload-eu-west-1-editorial-feeds-reuters_poller_lambda
      - lambda-upload-eu-west-1-editorial-feeds-apPoller_poller_lambda
      - asg-upload-eu-west-1-editorial-feeds-newswires
      - cfn-eu-west-1-editorial-feeds-wires-feeds
      - cfn-us-east-1-editorial-feeds-newswires-cloudfront-certificate
  lambda-update-eu-west-1-editorial-feeds-fingerpost-queueing-lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: fingerpost-queueing-lambda
    contentDirectory: fingerpost-queueing-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: fingerpost-queueing-lambda.zip
    actions:
      - updateLambda
    dependencies:
      - cfn-eu-west-1-editorial-feeds-newswires
  lambda-update-eu-west-1-editorial-feeds-ingestion-lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: ingestion-lambda
    contentDirectory: ingestion-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: ingestion-lambda.zip
    actions:
      - updateLambda
    dependencies:
      - cfn-eu-west-1-editorial-feeds-newswires
  lambda-update-eu-west-1-editorial-feeds-reingestion-initiation-lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: reingestion-initiation-lambda
    contentDirectory: reingestion-initiation-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: reingestion-initiation-lambda.zip
    actions:
      - updateLambda
    dependencies:
      - cfn-eu-west-1-editorial-feeds-newswires
  lambda-update-eu-west-1-editorial-feeds-email-filter-lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: email-filter-lambda
    contentDirectory: email-filter-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: email-filter-lambda.zip
    actions:
      - updateLambda
    dependencies:
      - cfn-eu-west-1-editorial-feeds-newswires
  lambda-update-eu-west-1-editorial-feeds-cleanup-lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: cleanup-lambda
    contentDirectory: cleanup-lambda
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: cleanup-lambda.zip
    actions:
      - updateLambda
    dependencies:
      - cfn-eu-west-1-editorial-feeds-newswires
  lambda-update-eu-west-1-editorial-feeds-reuters_poller_lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: reuters_poller_lambda
    contentDirectory: poller-lambdas
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: poller-lambdas.zip
    actions:
      - updateLambda
    dependencies:
      - cfn-eu-west-1-editorial-feeds-newswires
  lambda-update-eu-west-1-editorial-feeds-apPoller_poller_lambda:
    type: aws-lambda
    stacks:
      - editorial-feeds
    regions:
      - eu-west-1
    app: apPoller_poller_lambda
    contentDirectory: poller-lambdas
    parameters:
      bucketSsmLookup: true
      lookupByTags: true
      fileName: poller-lambdas.zip
    actions:
      - updateLambda
    dependencies:
      - cfn-eu-west-1-editorial-feeds-newswires
  asg-update-eu-west-1-editorial-feeds-newswires:
    type: autoscaling
    actions:
      - deploy
    regions:
      - eu-west-1
    stacks:
      - editorial-feeds
    app: newswires
    parameters:
      bucketSsmLookup: true
      prefixApp: true
    dependencies:
      - cfn-eu-west-1-editorial-feeds-newswires
    contentDirectory: newswires
  cfn-us-east-1-editorial-feeds-newswires-cloudfront-certificate:
    type: cloud-formation
    regions:
      - us-east-1
    stacks:
      - editorial-feeds
    app: newswires-cloudfront-certificate
    contentDirectory: cdk.out
    parameters:
      templateStagePaths:
        CODE: NewswiresCloudFrontCertificate-CODE.template.json
        PROD: NewswiresCloudFrontCertificate-PROD.template.json
"
`;
